sql/tsql/TSqlParser.g4

```diff
@@ -2879,68 +2961,67 @@ subquery
 
 // https://msdn.microsoft.com/en-us/library/ms175972.aspx
 with_expression
-    : WITH (XMLNAMESPACES ',')? common_table_expression (',' common_table_expression)*
-    | WITH BLOCKING_HIERARCHY ('(' full_column_name_list ')')? AS '(' select_statement ')'
+    : WITH ctes+=common_table_expression (',' ctes+=common_table_expression)*
     ;
 
 common_table_expression
-    : expression_name=id ('(' column_name_list ')')? AS '(' select_statement ')'
+    : expression_name=id ('(' columns=column_name_list ')')? AS '(' cte_query=select_statement ')'
     ;
 
 update_elem
-    : (full_column_name | LOCAL_ID) ('=' | assignment_operator) expression
+    : LOCAL_ID '=' full_column_name ('=' | assignment_operator) expression //Combined variable and column update
+    | (full_column_name | LOCAL_ID) ('=' | assignment_operator) expression
     | udt_column_name=id '.' method_name=id '(' expression_list ')'
     //| full_column_name '.' WRITE (expression, )
     ;
 
-// https://msdn.microsoft.com/en-us/library/ms173545.aspx
-search_condition_list
-    : search_condition (',' search_condition)*
+update_elem_merge
+    : (full_column_name | LOCAL_ID) ('=' | assignment_operator) expression
+    | udt_column_name=id '.' method_name=id '(' expression_list ')'
+    //| full_column_name '.' WRITE (expression, )
     ;
 
 search_condition
-    : search_condition_and (OR search_condition_and)*
-    ;
-
-search_condition_and
-    : search_condition_not (AND search_condition_not)*
+    : pred+=predicate_br (log=(OR | AND) pred+=predicate_br)*
     ;
 
-search_condition_not
-    : NOT? predicate
+predicate_br
+    : NOT* predicate
+    | NOT* '(' search_condition ')'
     ;
-
+    
 predicate
     : EXISTS '(' subquery ')'
+    | freetext_predicate
     | expression comparison_operator expression
     | expression comparison_operator (ALL | SOME | ANY) '(' subquery ')'
-    | expression NOT? BETWEEN expression AND expression
-    | expression NOT? IN '(' (subquery | expression_list) ')'
-    | expression NOT? LIKE expression (ESCAPE expression)?
+    | expression NOT* BETWEEN expression AND expression
+    | expression NOT* IN '(' (subquery | expression_list) ')'
+    | expression NOT* LIKE expression (ESCAPE expression)?
```

Comment:
It should be merged to the single alternative since all operators `BETWEEN`, `IN`, and `LIKE` have the same priority, see https://docs.microsoft.com/en-us/sql/t-sql/language-elements/operator-precedence-transact-sql?view=sql-server-ver15

```suggestion
    | expression NOT* (BETWEEN expression AND expression | IN '(' (subquery | expression_list) ')' | LIKE expression (ESCAPE expression)?)
```